bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonDeviceHandler.java

```diff
@@ -136,18 +136,27 @@ public void initialize() {
         config = getConfigAs(InsteonDeviceConfiguration.class);
 
         scheduler.execute(() -> {
-            if (getBridge() == null) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
                 String msg = "An Insteon network bridge has not been selected for this device.";
-                logger.warn("{} {}", thing.getUID().getAsString(), msg);
+                logger.debug("{} {}", thing.getUID().getAsString(), msg);
 
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
                 return;
             }
 
+            if (!ThingStatus.ONLINE.equals(bridge.getStatus())) {
+                String msg = String.format("Insteon network is not online");
+                logger.debug("{} {}", thing.getUID().getAsString(), msg);
+
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, msg);
```

Comment:
Since bridge is not **OFFLINE** (but UNKNOWN):
```suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, msg);
```

Reply:
But now not sure if framework is only marking child things with **BRIDGE_OFFLINE** as ONLINE when bridge comes online, or if detail is not considered. Can you test if this change would break the transition to ONLINE?

Reply:
Indeed:
https://github.com/openhab/openhab-core/blob/86ee4dc2b5802cab3fc3d7a575847e7ab40314d4/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/BaseThingHandler.java#L645-L647

Reply:
Resolving this comment as it would break the current fix.